bookstore/archive.py

```diff
@@ -1,45 +1,67 @@
+"""Archival of notebooks"""
 import json
-import aiobotocore
-
 from asyncio import Lock
 from typing import Dict, NamedTuple
 
+import aiobotocore
 from notebook.services.contents.filemanager import FileContentsManager
 from tornado import ioloop
 
 from .bookstore_config import BookstoreSettings
-
 from .s3_paths import s3_key, s3_display_path
 
 
 class ArchiveRecord(NamedTuple):
+    """Representation of archived content"""
     filepath: str
     content: str
     queued_time: float
 
 
 class BookstoreContentsArchiver(FileContentsManager):
-    """
-    Archives notebooks to S3 on save
-    """
+    """Archives notebooks to S3 on save by UI or parameterized workflow
+
+    Bookstore settings and Jupyter application settings are shared
+    A session is created for the current event loop
+    To write to a particular path on S3, acquire a path_lock.
+    The lock avoids multiple writes contending to write at the same time.
 
+    After acquiring the lock, `archive` authenticates using the storage service's
+    credentials. If allowed, the notebook is written to storage (i.e. S3).
+    """
     def __init__(self, *args, **kwargs):
         super(FileContentsManager, self).__init__(*args, **kwargs)
+
         # opt ourselves into being part of the Jupyter App that should have Bookstore Settings applied
         self.settings = BookstoreSettings(parent=self)
 
-        self.log.info("Archiving notebooks to {}".format(s3_display_path(
-            self.settings.s3_bucket, self.settings.workspace_prefix)))
+        self.log.info(
+            "Archiving notebooks to {}".format(
+                s3_display_path(self.settings.s3_bucket, self.settings.workspace_prefix)
+            )
+        )
 
-        self.session = aiobotocore.get_session()
+        # TODO: create a better check for credentials - keys and config or it will fail
+        try:
+            # create a session object from the current event loop
+            self.session = aiobotocore.get_session()
+        except Exception as e:
+            self.log.error(
+                "{} : Check credentials and configuration".format(e)
+            )
 
         # A lock per path to suppress writing to S3 when currently writing
         self.path_locks: Dict[str, Lock] = {}
         # Since locks are not pre-created,
         self.ability_to_create_path_lock = Lock()
```

Comment:
Can we rename `ability_to_create_path_lock` to `path_lock_aware` or `path_lock_capable`?